functions/executor/user_library.py

```diff
@@ -92,7 +101,16 @@ def _recv_inbox_listener(self):
         # Socket for receiving send() messages from other nodes.
         recv_inbox_socket = self.ctx.socket(zmq.PULL)
         recv_inbox_socket.bind(server_utils.BIND_ADDR_TEMPLATE % (server_utils.RECV_INBOX_PORT + self.executor_tid))
+        t = threading.currentThread()
 
-        while True:
-            (sender, msg) = recv_inbox_socket.recv_pyobj(0)  # Blocking.
-            self.recv_inbox.put((sender, msg))
+        while t.do_run:
+            try:
+                (sender, msg) = recv_inbox_socket.recv_pyobj(zmq.NOBLOCK)
```

Comment:
Do we need to do a sleep() here? Should we be worried about this busy looping consuming cycles from the function execution, since it runs in the same process?

Reply:
Hm. Good question. I think it should be fine for now. I'm not an expert on how Python will do multithreading to be honest. You might know better than me.